operators/src/processing/reprojection.rs

```diff
@@ -416,10 +416,20 @@ where
     ) -> Result<BoxStream<'a, Result<geoengine_datatypes::raster::RasterTile2D<Self::RasterType>>>>
     {
         // we need a resolution for the sub-querys. And since we don't want this to change for tiles, we precompute it for the complete bbox and pass it to the sub-query spec.
-        // TODO: use `rewrite_query` to determine resolution for tiles overlapping border
-        let projector = CoordinateProjector::from_known_srs(self.to, self.from)?;
+        let projector_source_target = CoordinateProjector::from_known_srs(self.from, self.to)?;
+        let projector_target_source = CoordinateProjector::from_known_srs(self.to, self.from)?;
+
+        let p_bbox = query.bbox.reproject_clipped(&projector_target_source)?;
+        let s_bbox = p_bbox.reproject(&projector_source_target)?;
+
         let p_spatial_resolution =
-            suggest_pixel_size_from_diag_cross(query.bbox, query.spatial_resolution, &projector)?;
+            suggest_pixel_size_from_diag_cross_projected(s_bbox, p_bbox, query.spatial_resolution)?;
```

Comment:
I am not sure that a constant resolution for querying the source is desirable. We CAN do better, why shouldn't we?

Reply:
if you change the query rectangle it will request a different resolution. this will be a problem when we have a cache for tiles. with this approach, the tile resolution will change for every query and will render a cache useless.

Reply:
> this strategy will produce different results for the same area when it is covered by non-eequal boundingboxes.

If we use a global resolution I think we will have to use the finest resolution everywhere. Because otherwise we cant get a certain area in the best quality. Maybe a dataset is only defined in one part of the map where we need the highest resolution for the projection but calculating the resoultion for the whole projection doesn't get us this. What do you think?

I agree that we should produce the same tiles, regardless of the total query bbox. Would it be an acceptable compromise to compute the resolution for each tile of the global grid? Then it would be reproducable independently of the query rectangle but still sensitive to the local changes in the resolution across the map.

Reply:
yes would work for me. however, i would suggest we calculate it for the tiles and use the finest for all